src/types/closure.md

```diff
@@ -264,6 +267,69 @@ let c = || {
 
 [wildcard pattern]: ../patterns.md#wildcard-pattern
 
+r[type.closure.capture.precision.discriminants]
+### Capturing for discriminant reads
+
+If pattern matching requires inspecting a discriminant, the relevant place will get captured by `ImmBorrow`.
```

Comment:
Unless we mutably access the contents, right? E.g.
```rust
let c = || match x {
    (Example::A(ref mut _y), _) => println!("variant A"),
    (Example::B(_), _) => println!("variant B"),
};
```

Reply:
ok yep, that's the "Shared prefix" thing

Reply:
Yup. Do you think this deserves a clarification? (if so, how would you word it)